neutron/convert.cpp

```diff
@@ -271,4 +273,43 @@ Dataset convert(Dataset d, const Dim from, const Dim to) {
   // MDZipView<const Coord::TwoTheta>(dataset);
 }
 
+
+//using namespace scipp::core::except::expect;
+
+//namespace scipp::neutron::diffraction {
+namespace diffraction {
```

Comment:
I think file structure should follow namespace structure here. I suggest to put `convert_with_calibration` into a subfolder `diffraction`. This will also help people to know what to include `scipp/neutron/diffraction/convert_with_calibration.h` to get `scipp::neutron::diffraction::convert_with_calibration()`.

Reply:
Would you want the header file with the cpp file in the diffraction subfolder, or in the include folder as with convert?
scipp/neutron/diffraction/convert_with_calibration.h
or
scipp/neutron/include/scipp/neutron/diffraction/convert_with_calibration.h?

Reply:
Everything that is "public" (as in, all headers that need to be in an installed `include/` folder, for someone installing the c++ library) has to be in `scipp/neutron/include/scipp/neutron` or a subfolder, so the latter.

We are currently not actually distributing/shipping the C++ version of the library, but eventually this will become relevent.